client.go

```diff
@@ -150,17 +145,8 @@ func newClient() (*client, error) {
 
 // Close is used to cleanup the client
 func (c *client) Close() error {
-	c.closeLock.Lock()
-	defer c.closeLock.Unlock()
-
-	if c.closed {
-		return nil
-	}
-	c.closed = true
-
-	log.Printf("[INFO] mdns: Closing client %v", *c)
+	log.Printf("[INFO] mdns: Closing client: %v", *c)
 	close(c.closedCh)
```

Comment:
FWIW, `Close()` will now panic if called twice. Before, it would simply return `nil` after second call.